src/CloudIoTCoreDevice.cpp

```diff
@@ -0,0 +1,118 @@
+/******************************************************************************
+ * Copyright 2018 Google
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+#include "CloudIoTCoreDevice.h"
+#include "Arduino.h"
+#include "jwt.h"
+
+CloudIoTCoreDevice::CloudIoTCoreDevice()
+{
+}
+
+CloudIoTCoreDevice::CloudIoTCoreDevice(const char *project_id, const char *location, const char *registry_id, const char *device_id)
+{
+    setProjectId(project_id);
+    setLocation(location);
+    setRegistryId(registry_id);
+    setDeviceId(device_id);
+}
+
+CloudIoTCoreDevice::CloudIoTCoreDevice(const char *project_id, const char *location, const char *registry_id, const char *device_id, const char *private_key)
+{
+    setProjectId(project_id);
+    setLocation(location);
+    setRegistryId(registry_id);
+    setDeviceId(device_id);
+    setPrivateKey(private_key);
+}
+
+std::string CloudIoTCoreDevice::createJWT(long long int current_time)
+{
+    jwt = CreateJwt(project_id, current_time, priv_key);
+    return jwt;
+}
+
+std::string CloudIoTCoreDevice::getJWT()
```

Comment:
avoid std::string -- as before, we should prefer String over std::string because it's a better object type for Arduino.

Reply:
I'll change all to `String` instead of `std::string`.